_posts/2023-02-13-changelog-188.markdown

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #188"
+date: 2023-02-13 13:00:00 +0300
+---
+
+
+## New Features
+
+* Detect new compiler errors (by [@kuksag]):
+  * [#10066] `main` function is not allowed to have generic parameters ([E0131](https://doc.rust-lang.org/error-index.html#E0131))
+
+  * [#10066] An inherent implementation was marked `unsafe` ([E0197](https://doc.rust-lang.org/error-index.html#E0197))
+
+  * [#10066] Type parameter has more than one relaxed default bound ([E0203](https://doc.rust-lang.org/error-index.html#E0203))
+
+  * [#9894] A pattern was declared as an argument in a foreign function declaration ([E0130](https://doc.rust-lang.org/error-index.html#E0130))
+
+  * [#9894] Trait methods currently cannot take patterns as arguments ([E0642](https://doc.rust-lang.org/error-index.html#E0642))
+
+* [#10057] Parse and annotate [`const_closures`](https://github.com/rust-lang/rust/issues/106003) syntax
+
+* [#8826] Support `where` clauses [on associated types](https://rust-lang.github.io/rfcs/1598-generic_associated_types.html#where-clauses-on-associated-types)
+
+* [#9987] Annotate missing lifetimes ([E0106](https://doc.rust-lang.org/error-index.html#E0106)) in function signatures
+
+* [#9123] Allow excluding specific paths from auto-import via `Preferences | Editor | General | Auto Import | Rust` settings.
+By default, [`Borrow`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) and [`BorrowMut`](https://doc.rust-lang.org/std/borrow/trait.BorrowMut.html) methods are excluded, as auto-import is rarely desired for them.
+Tip: exclude functions or traits that are re-exported in irrelevant crates to get only the original one in the suggestions.
+For further details, please refer to the pull request description
+
+  <img src="/assets/posts/changelog-188/exclude_paths_auto_import.png" width="600px"/>
+
+* [#7489] Add debugger option to not step into stdlib sources.
+Note, there is a [known limitation](https://github.com/intellij-rust/intellij-rust/issues/10088) that it does not work yet on Windows with MSVC toolchain (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-188/do_not_step_into_stdlib" w="600px" %}
```

Comment:
Are you sure that to use new UI for this screenshot? Others (except new UI icons) use old one

Reply:
My idea was to start promoting the new UI since it is more or less supported by our plugin.
Speaking of consistency, I can remake other screenshots if the difference catches the eye

Reply:
It's a good idea in general.
But I'd use old style for now since not all icons are updated yet and some of them are under discussion

Reply:
OK, I remade `do_not_step_into_stdlib` gif